torchdynamo/debug_utils.py

```diff
@@ -327,24 +372,24 @@ def dump_to_minify(gm, args, compiler_name: str):
     return helper_for_dump_minify(contents)
 
 
-def wrap_compiler_debug(compiler, compiler_name: str):
+def wrap_compiler_debug(compiler_fn, compiler_name: str):
     """
     Minifier for Fx Graph modules after Aot Autograd has finished. We wrap both
-    forward and backward call separately with the backend compiler - like
+    forward and backward call separately with the backend compiler_fn - like
     inductor or nvfuser. Intercepting after Aot Autograd presents neat
     abstration, where all the params are lifted as graph inputs, making it easy
     to save the graph as a string.
     """
 
-    @functools.wraps(compiler)
+    @functools.wraps(compiler_fn)
     def debug_wrapper(gm, example_inputs, **kwargs):
         orig_graph = copy.deepcopy(gm.graph)
         assert config.repro_after in ("dynamo", "aot", None)
 
         def deferred_for_real_inputs(*real_inputs):
```

Comment:
No need to fix, but this seems kinda ... strange to me. We delay compilation from the ... first time we see inputs, to slightly later after the first time we see inputs.

Reply:
We could do compilation with fake tensors first and then run the compiled model with real tensors later. But the code gets real ugly.